test/danetest.c

```diff
@@ -182,18 +159,16 @@ static char *read_to_eol(BIO *f)
         return NULL;
 
     n = strlen(buf);
-
-    if (buf[n-1] != '\n') {
-        if (n+1 == sizeof(buf)) {
-            fprintf(stderr, "%s: warning: input too long\n", progname);
-        } else {
-            fprintf(stderr, "%s: warning: EOF before newline\n", progname);
-        }
+    if (buf[n - 1] != '\n') {
+        if (n + 1 == sizeof(buf))
+            TEST_error("input too long");
+        else
+            TEST_error("EOF before newline");
```

Comment:
The ```read_to_eol``` is called by ```test_tlsafile``` and it does not return error, only NULL on EOF.  So an error condition must be signaled inside ```read_to_eol``` by callind ```TEST_error``` directly.
In the other instances you mentioned, they call ```TEST_info``` because their calling function sees an error return and then calls ```TEST_error``` themselves.

Make sense?

Yes it's complicated.  I did not write this code :)  I just converted it.